### PR TITLE
ascanrules: Add SSTI example alerts

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
- CHANGELOG > Add notes.
- Scan rules > Add example alert functionality (6119), add appropriate alert tags.
- Unit tests > Assert the new example alerts, and mappings.

## Related Issues
- zaproxy/zaproxy#6119

## Checklist
- [NA] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

### CODE REVIEW

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SstiBlindScanRule.java [hunk 1]
```diff
@@ -368,4 +380,35 @@ private void sendPayloadsToMakeCallBack(String paramName, String[] commandExecPa
             }
         }
     }
+
+    AlertBuilder createAlert(String url, String param, String attack, HttpMessage msg) {
```
Comment (thc202) [inline addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SstiBlindScanRule.java:384]: `private`

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SstiBlindScanRule.java [hunk 2]
```diff
@@ -368,4 +380,35 @@ private void sendPayloadsToMakeCallBack(String paramName, String[] commandExecPa
...
+        return newAlert()
+                .setConfidence(Alert.CONFIDENCE_HIGH)
+                .setUri(url)
+                .setParam(param)
+                .setAttack(attack)
+                .setMessage(msg);
```
Comment (thc202) [inline addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SstiBlindScanRule.java:390]: Let the caller do this and avoid the need to create a message for the example. Same for the other scan rule.

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SstiBlindScanRule.java [hunk 3]
```diff
@@ -368,4 +380,35 @@ private void sendPayloadsToMakeCallBack(String paramName, String[] commandExecPa
...
+    public List<Alert> getExampleAlerts() {
+        HttpMessage msg = null;
+        try {
+            msg = new HttpMessage(new URI("http://example.com/profile/?name=foo", true));
+        } catch (URIException | HttpMalformedHeaderException | NullPointerException e) {
+            // Ignore
+        }
+        return List.of(
+                createAlert(
+                                msg.getRequestHeader().getURI().toString(),
```
Comment (thc202) [inline addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SstiBlindScanRule.java:403]: `"http://example.com/profile/?name=foo"`

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SstiScanRule.java [hunk 4]
```diff
@@ -415,4 +420,58 @@ private void searchForMathsExecution(
...
+                        + "            <input type=\"submit\" value=\"Submit\">\n"
+                        + "        </form>\n"
+                        + "        <h2>Hello zj3790300zj</h2>\n"
+                        + "    </body>\n"
+                        + "</html>Content-Type: text/html\n"
+                        + "Date: Mon, 10 Jun 2024 12:33:36 GMT\n"
+                        + "Connection: keep-alive\n"
+                        + "Content-Length: 328\n";
+        String otherInfo =
+                Constant.messages.getString(
```
Comment (thc202) [inline addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SstiScanRule.java:457]: This should be extracted.

**Comment by kingthorin** [comment]
Got all those I think.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**